### PR TITLE
operations: better handling of range requests

### PR DESCRIPTION
**Pull request by pyr**
This brings the following changes to range request handling:

- Return HTTP 416 "Unsatisfiable Range" for invalid ranges.
- Return HTTP 206 "Partial Content" for valid range responses.
- Bypassing range handling for full file requests.
- Fix range requests that span beyond the first chunk.

### CODE REVIEW

**Comment by pyr** [comment]
This also updates the test suite to catch a larger family of logic errors.

File: src/io/pithos/operations.clj [hunk 1]
```diff
@@ -125,13 +125,26 @@
...
+(defn check-range
+  [max [has-range? start end]]
+  (let [e (ex-info "unsatisfiable range"
+                   {:type :invalid-argument
+                    :arg "range"
+                    :val (format "start=%s,end=%s" start end)
+                    :status-code 416})]
+    (cond
+      (> start end) (throw e)
+      :else         [has-range? start end])))
```
Comment (sfriesel) [inline src/io/pithos/operations.clj]: shouldn't this be `:else max`?

Comment (pyr) [inline src/io/pithos/operations.clj]: here i'm only checking that the provided range makes sense. if it does I return what I was given.
The destructuring trick I could have used:

```clojure
(defn check-range
   [max [_ start end :as range]]
     ....
        :else range)
```

**Pull request merged**